### ISSUE TITLE
IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager

### ISSUE THREAD

**Issue by Strubbl** [square/okhttp#6069]
since release 4.7.0 and also for 4.7.1 i get the following exception when using the OkHttpClient.

```
2020-05-19 13:15:39.061 13971-14060/fr.gaulupeau.apps.InThePoche.debug E/ArticleUpdateWorker: updateArticles() exception
    java.lang.IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager
        at org.conscrypt.Conscrypt.toConscrypt(Conscrypt.java:736)
        at org.conscrypt.Conscrypt.setHostnameVerifier(Conscrypt.java:769)
        at okhttp3.internal.platform.ConscryptPlatform.platformTrustManager(ConscryptPlatform.kt:53)
        at okhttp3.OkHttpClient.<init>(OkHttpClient.kt:237)
        at okhttp3.OkHttpClient$Builder.build(OkHttpClient.kt:1069)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createClient(WallabagConnection.java:123)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createWallabagService(WallabagConnection.java:107)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.getWallabagService(WallabagConnection.java:38)
        at fr.gaulupeau.apps.Poche.service.workers.BaseNetworkWorker.getWallabagService(BaseNetworkWorker.java:86)
        at fr.gaulupeau.apps.Poche.service.workers.ArticleUpdateWorker.updateArticles(ArticleUpdateWorker.java:76)
        at fr.gaulupeau.apps.Poche.service.workers.ArticleUpdateWorker.update(ArticleUpdateWorker.java:37)
        at fr.gaulupeau.apps.Poche.service.tasks.UpdateArticlesTask.run(UpdateArticlesTask.java:17)
        at fr.gaulupeau.apps.Poche.service.tasks.ActionRequestTask.run(ActionRequestTask.java:25)
        at fr.gaulupeau.apps.Poche.service.-$$Lambda$Eg7UQkyPgCNRHvigqxzb-ridugA.run(Unknown Source:2)
        at fr.gaulupeau.apps.Poche.service.TaskService.run(TaskService.java:158)
        at fr.gaulupeau.apps.Poche.service.TaskService.lambda$gjfracnqY8x0_-mrxsx0oqjQmgk(Unknown Source:0)
        at fr.gaulupeau.apps.Poche.service.-$$Lambda$TaskService$gjfracnqY8x0_-mrxsx0oqjQmgk.run(Unknown Source:2)
        at java.lang.Thread.run(Thread.java:919)
```

We create the client like this (simplified from [here](https://github.com/wallabag/android-app/blob/master/app/src/main/java/fr/gaulupeau/apps/Poche/network/WallabagConnection.java#L107)):

```
OkHttpClient.Builder b = new OkHttpClient.Builder()
                .readTimeout(45, TimeUnit.SECONDS);
b.build();
```

This issue does not occur with release 4.6.0 or 4.5.0.

Is this a bug? Or do you have an idea how i can fix this exception?

**Comment by di72nn** [comment]
Some additional info: the project is an Android application which uses the Conscrypt library.

Minimal case:
```gradle
    implementation 'org.conscrypt:conscrypt-android:2.4.0'
    implementation 'com.squareup.okhttp3:okhttp:4.7.1'
```

```java
        Security.insertProviderAt(Conscrypt.newProvider(), 1);

        new OkHttpClient();
```

**Comment by yschimke** [comment]
Our 4.x Android tests use `Security.insertProviderAt(Conscrypt.newProviderBuilder().build(), 1)`

https://github.com/square/okhttp/blob/master/android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt#L165

https://github.com/square/okhttp/blob/master/android-test/build.gradle#L50

**Comment by yschimke** [comment]
The core issue here is the Platform check seems to be failing and we are trying to use Conscrypt JVM provider on Android.

**Comment by yschimke** [comment]
Is this trivially reproducible?  Can you provide details about your test devices? Or which Android versions + devices you see this in Prod?

**Comment by yschimke** [comment]
This project provides a trivial repro of the issue.  It looks like something kotlin/proguard/dalvik related.  Debugging makes no sense.

https://github.com/wallabag/android-app

<img width="1262" alt="image" src="https://user-images.githubusercontent.com/231923/82411524-d68c8b80-9a69-11ea-9c12-fcde3066d728.png">

**Comment by swankjesse** [comment]
Released in 4.7.2.
https://square.github.io/okhttp/changelog/#version-472
